src/components/elements/controlledDisplayFiltersGroup/index.tsx

```diff
@@ -23,28 +23,71 @@ export function ControlledDisplayFiltersGroup({
   label
 }: ControlledDisplayFiltersGroupProps) {
   const [showAll, setShowAll] = useState<boolean>(false);
+  const [displayedFilters, setDisplayedFilters] = useState<Set<string>>(() => new Set<string>());
+
+  const availableFiltersNames= new Set(filters.map((filter)=>filter.label));
 
   const toggleAll = () => {
+    if(!showAll){
+      setDisplayedFilters(availableFiltersNames);
+    }
+    else{
+      setDisplayedFilters(new Set());
+    }
     setShowAll(!showAll);
     toggleShowAll(!showAll);
   };
 
-  const updateFilter = (name: string, show: boolean) => {
-    onFilterChange(name, show);
+  function setsAreEqual(setA:Set<string>, setB:Set<string>) {
+    if (setA.size !== setB.size) {
+      return false;
+    }
+    for (const item of setA) {
+      if (!setB.has(item)) {
+        return false;
+      }
+    }
+    return true;
+  }
```

Comment:
1. If we're already using the lodash lib, it has this function built in https://stackoverflow.com/a/36866642
2. If not, then this is an extremely generic function, I would make it accept any input type and put it in some general "utils" module that you can then import in other places. 
```typescript
function areSetsEqual<T>(setA: Set<T>, setB: Set<T>): boolean {
  // Check if the sizes are different
  if (setA.size !== setB.size) return false;

  // Check if all elements of setA are in setB
  for (const element of setA) {
    if (!setB.has(element)) return false;
  }

  return true;
}
```

Reply:
done